src/engraving/libmscore/measure.cpp

```diff
@@ -4115,12 +4077,77 @@ static bool hasAccidental(Segment* s)
 }
 
 //---------------------------------------------------------
-//   computeMinWidth
-//    sets the minimum stretched width of segment list s
-//    set the width and x position for all segments
+//      minSysTicks
+//      returns the shortest note/rest in the system
+//---------------------------------------------------------
+
+Fraction Measure::minSysTicks()
+{
+    Fraction minTicks = Fraction(10, 1);
+    for (MeasureBase* mb : system()->measures()) {
+        if (mb->isMeasure()) {
+            Measure* m = toMeasure(mb);
+            Fraction curMinTicks = m->computeTicks();
+            if (curMinTicks < minTicks) {
+                minTicks = curMinTicks;
+            }
+        }
+    }
+    return minTicks;
+}
+
+//---------------------------------------------------------
+//      Stretch formula
+//      Computes the stretch for a given segment depending
+//      on its duration with respect to the shortest one.
+//      Three different options proposed (see documentation).
+//---------------------------------------------------------
+qreal Measure::stretchFormula(Fraction curTicks, Fraction minTicks, qreal stretchCoeff)
+{
+    qreal slope = userStretch() * score()->styleD(Sid::measureSpacing);
+
+    if (curTicks > m_timesig) { // This is the case of MM rests
+        curTicks = curTicks - m_timesig; // 2 bars MM rests is spaced same as 1 bar
+        if (curTicks > m_timesig * 20) { // To avoid excessive
```

Comment:
it's not clear what 20 does here, probably requires a comment or, again, named constant variable